packages/server/src/v1/pulse/infrastructure/database/__tests__/pulse.repo.test.ts

```diff
@@ -2,15 +2,125 @@ import { knex } from '../knex'
 import { PulseRepo } from '../pulse.repo'
 
 const pulseRepo = new PulseRepo(knex)
+
```

Comment:
Some of the behaviors of the report I think we should test:
- Does the query group many heartbeats within a two minute period. i.e. 12 heartbeats in 2 minute interval equals two minutes of time
- Are heartbeats outside of 2 minutes additive? if I get 1 heartbeat now, and 1 heartbeat in 10 minutes, do I get 4 minutes of time?
- Edge case: Does the query handle days. If I have a heartbeat at 11:59pm or 12:00am, does that heartbeat count towards the correct day?
- Does the query represent category time correctly? If I get a heartbeat for browsing, do I get 2 minutes for browsing?

Reply:
Will Update! 

I think part of my issues with writing tests here was I also don't know what the use cases are of this code. It helps to go through scenarios.

Reply:
@rphovley Added all except the category time correctly, since I don't think we have that logic yet. Put a TODO comment on there.